tests/e2e/volumepropverification_test.go

```diff
@@ -24,118 +24,122 @@ import (
-				vmNameFrmAdminCli := verification.GetVMAttachedToVolUsingAdminCli(volName, esx)
-				if dockerCliCheck {
-					vmNameFrmDockerCli := verification.GetVMAttachedToVolUsingDockerCli(volName, vms[vmIndx])
-					// TODO: get vm name based on ip and compare ith with the docker cli and admin cli
-					if vmNameFrmDockerCli != vmNameFrmAdminCli {
-						log.Fatalf("Information mis-match - Attached-to-VM field for volume from docker cli is [%s]"+
-							"and attched-to-vm field from admin cli is [%s]", vmNameFrmDockerCli, vmNameFrmAdminCli)
-					}
-					volumePropertiesDockerCli := verification.GetVolumePropertiesDockerCli(volName, vms[vmIndx])
-					expctdPropsDkr := []string{"100MB", formatTypes[k]}
-					if !hasElement(expctdPropsDkr, volumePropertiesDockerCli) {
-						log.Fatal("Volume properties on ESX fetched using docker cli do not matches with the expected values")
-					}
-				}
-				volmPropertiesAdminCli = verification.GetVolumePropertiesAdminCli(volName, esx)
-				expctdPropsAdmin = []string{"100MB", formatTypes[k]}
-				if !hasElement(expctdPropsAdmin, volmPropertiesAdminCli) {
-					log.Fatal("Volume properties on admin cli do not matches with the expected values")
-				}
-				log.Println("Finished verifying volume properties like size, disk-format and attached-to-vm fields"+
-					" at vm and esx for volume - ", volName)
-				ssh.InvokeCommand(vms[vmIndx], "docker stop "+containerName+" ; docker rm "+containerName)
-			}
+
+func (s *VolumePropVerificationSuite) TestVolumeProperties(c *C) {
+	log.Printf("START: TestVolumeProperties")
+	formatTypes := []string{"thin", "zeroedthick", "eagerzeroedthick"}
+	for i := 0; i < len(formatTypes); i++ {
+		s.volumeName = inputparams.GetVolumeNameWithTimeStamp(formatTypes[i])
+		s.containerName = inputparams.GetContainerNameWithTimeStamp("volumepropverification")
+
+		log.Printf("Verifying volume properties like size, disk-format and attached-to-vm fields "+
+			"at vm and esx for volume [%s] ", s.volumeName)
+		out, err := ssh.InvokeCommand(s.dockerHostIP, con.CreateVolume+"--name="+s.volumeName+" -o diskformat="+formatTypes[i])
+		c.Assert(err, IsNil, Commentf(out))
+
+		volumePropsAdminCli := verification.GetVolumePropertiesAdminCli(s.volumeName, s.esxIP)
+		expectedPropsAdminCli := []string{"100MB", formatTypes[i], "detached"}
+		c.Assert(hasElement(expectedPropsAdminCli, volumePropsAdminCli), Equals, true, Commentf("Volume %s properties "+
+			"on ESX fetched using admin cli does not matches with the expected values", s.volumeName))
+
+		volumePropsDockerCli := verification.GetVolumePropertiesDockerCli(s.volumeName, s.dockerHostIP)
+		expectedPropsDockerCli := []string{"100MB", formatTypes[i], "<no value>"}
+		c.Assert(hasElement(expectedPropsDockerCli, volumePropsDockerCli), Equals, true, Commentf("Volume %s properties "+
+			"on docker vm fetched using docker cli does not matches with the expected values", s.volumeName))
+
+		if formatTypes[i] == "thin" {
```

Comment:
1. Move this handling for thin disks into a separate function.

2. Add a new function that does "mandatory checks on volume properties"

3. Get the admin CLI and docker CLI volume properties and pass to function in (2) to make the checks.

4. If volume is thin then call another function to verify volume properties specific for thin disks.

Make this function smaller by breaking it up into separate functions

1. One function to create the volumes,
2. Then loop over the volumes and get the admin and docker CLI properties for each volume.
3. Call the property validation function for each volume.